dask/dataframe/io/demo.py

```diff
@@ -99,20 +249,37 @@ def __call__(self, part):
         divisions, state_data = part
         if isinstance(state_data, int):
             state_data = random_state_data(1, state_data)
-        return make_timeseries_part(
+        return make_dataframe_part(
+            self.index_dtype,
             divisions[0],
             divisions[1],
             self.dtypes,
             self.columns,
-            self.freq,
             state_data,
             self.kwargs,
         )
 
 
-def make_timeseries_part(start, end, dtypes, columns, freq, state_data, kwargs):
-    index = pd.date_range(start=start, end=end, freq=freq, name="timestamp")
+def make_dataframe_part(index_dtype, start, end, dtypes, columns, state_data, kwargs):
     state = np.random.RandomState(state_data)
+    if pd.api.types.is_datetime64_any_dtype(index_dtype):
```

Comment:
Just a note (no need to address here): This also includes timezone aware dtypes. This information would be lost in the date_range call